pkg/machine/qemu/machine_test.go

```diff
@@ -36,5 +82,26 @@ func TestPropagateHostEnv(t *testing.T) {
 	tokens := strings.Split(cmdLine[1], ",string=")
 	decodeString, err := base64.StdEncoding.DecodeString(tokens[1])
 	assert.NoError(t, err)
-	assert.Equal(t, fmt.Sprintf("HTTP_PROXY=\"proxy\"|SSL_CERT_FILE=\"%s/foo.cert\"|SSL_CERT_DIR=%q", machine.UserCertsTargetPath, machine.UserCertsTargetPath), string(decodeString))
+
+	// envsRawArr looks like: ["BAR=\"bar\"", "FOO=\"foo\""]
+	envsRawArr := strings.Split(string(decodeString), "|")
+	// envs looks like: {"BAR": "bar", "FOO": "foo"}
+	envs := make(map[string]string)
+	for _, env := range envsRawArr {
+		item := strings.Split(env, "=")
```

Comment:
Very nice! As a point of general paranoia, this might be safer as `.SplitN(env, "=", 2)` to protect against values containing an equal sign (super unlikely). IMO not worth fixing unless you need to re-push for other reasons; I just mention it as something to think about in the future. Thank you!